app/uk/gov/hmrc/personaldetailsvalidation/services/PersonalDetailsValidatorService.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.personaldetailsvalidation.services
+
+import akka.Done
+import cats.data.EitherT
+import uk.gov.hmrc.personaldetailsvalidation.PdvRepository
+import uk.gov.hmrc.personaldetailsvalidation.model.{PersonalDetailsValidation, ValidationId}
+
+import javax.inject.{Inject,Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PersonalDetailsValidatorService @Inject()(pdvRepository: PdvRepository){
+
+  def insertRecord(personalDetails: PersonalDetailsValidation)(implicit ec: ExecutionContext):
+  EitherT[Future, Exception, Done] = pdvRepository.create(personalDetails)
+
+  def getRecord(personalDetailsValidationId: ValidationId)(implicit ec: ExecutionContext):
+  Future[Option[PersonalDetailsValidation]] = pdvRepository.get(personalDetailsValidationId)
+
```

Comment:
I like that you added this
But currently the get record isnt called in the codebase,
I'd suggest refactoring everywhere that currently calls pdvRepository.get to be refactored with PersonalDetailsValidatorService.getRecord

Also can you write tests for both of these functions to keep coverage up

Reply:
refactored and methods are covered by tests

Reply:
still not covered, can we have a PersonalDetailsValidatorServiceSpec please covering these 2 functions

Reply:
PersonalDetailsValidatorServiceSpec has been added and covers the above methods